ReadbasedAnalysis.wdl

```diff
@@ -217,94 +221,90 @@ task finish_reads {
-        mkdir -p ${outdir}/kraken2
-        cp ${kraken2_classification_tsv} ${kraken2_report_tsv} ${kraken2_krona_html} \
-           ${outdir}/kraken2
+        mkdir -p ~{outdir}/gottcha2
+        cp ~{gottcha2_report_tsv} ~{gottcha2_full_tsv} ~{gottcha2_krona_html} \
+           ~{outdir}/gottcha2
+        mkdir -p ~{outdir}/centrifuge
+        cp ~{centrifuge_classification_tsv} ~{centrifuge_report_tsv} ~{centrifuge_krona_html} \
+           ~{outdir}/centrifuge
+        mkdir -p ~{outdir}/kraken2
+        cp ~{kraken2_classification_tsv} ~{kraken2_report_tsv} ~{kraken2_krona_html} \
+           ~{outdir}/kraken2
     >>>
     runtime {
         docker: container
         memory: "1 GiB"
         cpu:  1
     }
     output{
-        Array[String] fastq_files = glob("${outdir}/*.fastq*")
+        Array[String] fastq_files = glob("~{outdir}/*.fastq*")
     }
 }
 
 
 task make_info_file {
-    Boolean enabled_tools_gottcha2
-    Boolean enabled_tools_kraken2
-    Boolean enabled_tools_centrifuge
-    String db_gottcha2
-    String db_kraken2
-    String db_centrifuge
-    String docker
-    File? gottcha2_report_tsv
-    File? gottcha2_info
-    File? centrifuge_report_tsv
-    File? centrifuge_info
-    File? kraken2_report_tsv
-    File? kraken2_info
-    String info_filename = "profiler.info"
-
+    input {
+        String db_gottcha2
+        String db_kraken2
+        String db_centrifuge
+        String docker
+        File gottcha2_report_tsv
+        File gottcha2_info
+        File centrifuge_report_tsv
+        File centrifuge_info
+        File kraken2_report_tsv
+        File kraken2_info
+        String info_filename = "profiler.info"
+    }
     command <<<
         set -euo pipefail
 
         # generate output info file
 
         info_text="Taxonomy profiling tools and databases used: "
-        echo $info_text > ${info_filename}
-
-        if [[ ${enabled_tools_kraken2} == true ]]
-        then
-            software_ver=`cat ${kraken2_info}`
-            #db_ver=`echo "${db_kraken2}" | rev | cut -d'/' -f 1 | rev`
-            db_ver=`cat ${db_kraken2}/db_ver.info`
-            info_text="Kraken2 v$software_ver (database version: $db_ver)"
-            echo $info_text >> ${info_filename}
-        fi
+        echo $info_text > ~{info_filename}
 
-        if [[ ${enabled_tools_centrifuge} == true ]]
-        then
-            software_ver=`cat ${centrifuge_info}`
-            db_ver=`cat $(dirname ${db_centrifuge})/db_ver.info`
-            info_text="Centrifuge v$software_ver (database version: $db_ver)"
-            echo $info_text >> ${info_filename}
-        fi
 
-        if [[ ${enabled_tools_gottcha2} == true ]]
-        then
-            software_ver=`cat ${gottcha2_info}`
-            db_ver=`cat $(dirname ${db_gottcha2})/db_ver.info`
-            info_text="Gottcha2 v$software_ver (database version: $db_ver)"
-            echo $info_text >> ${info_filename}
-        fi
+        software_ver=`cat ~{kraken2_info}`
```

Comment:
I do think we still want the if logic in case certain components are not run. There should also be a reporting block for singleM.